### PR TITLE
tests.e2e.ansible: Fix to increase github rate limiting for kustomize

### PR DESCRIPTION
**Pull request by GabyCT**
This PR authenticates for github to increase the threshold for rate limiting while installing kustomize and avoid the random failures that are impacting several CIs.

### CODE REVIEW

**Comment by GabyCT** [comment]
I tried this PR locally on the TDX machine putting the installation of kustomize in a for loop and seems to be working

**Comment by GabyCT** [comment]
@wainersm is there a way to test this on tdx and other CIs?

File: tests/e2e/ansible/install_test_deps.yaml [hunk 1]
```diff
@@ -39,7 +39,7 @@
       ignore_errors: yes
     - name: Install kustomize
       shell: |
-        curl -s --retry 3 --retry-delay 10 "https://raw.githubusercontent.com/kubernetes-sigs/kustomize/master/hack/install_kustomize.sh"  | bash
+        curl -s --retry 3 --retry-delay 10 -u ${GITHUB_USER}:${GITHUB_TOKEN} "https://raw.githubusercontent.com/kubernetes-sigs/kustomize/master/hack/install_kustomize.sh"  | bash
```
Comment (ldoktor) [inline tests/e2e/ansible/install_test_deps.yaml:42]: IIRC the GITHUB_USER and GITHUB_TOKEN is not set by default in the CI workflows and one has to specifically set it via env: https://docs.github.com/en/rest/authentication/authenticating-to-the-rest-api?apiVersion=2022-11-28 or am I wrong? (note curl will happily accept `-u :` so it should work well with empty user as well.

Comment (GabyCT) [reply tests/e2e/ansible/install_test_deps.yaml:42]: @ldoktor locally if I put echo $GITHUB_USER and $GITHUB_TOKEN seems to exists on the TDX machine

Comment (ldoktor) [reply tests/e2e/ansible/install_test_deps.yaml:42]: Well it's not in https://docs.github.com/en/actions/learn-github-actions/variables#default-environment-variables so might I ask you to add it to the workflows as well to ensure it's going to work properly on all workers?

**Comment by GabyCT** [comment]
Is this going to test the modification of the workflow GHA yaml?

**Review by ldoktor — APPROVED** [review]
Thanks, lgtm now. I was wondering a bit about the `${USER}` but I found out github ignores it whatsoever and simply uses `token` (in documentation they actually mention one can use `Bearer` or `token` but I tried random characters and it's perfectly happy with it. Another good surprise for me was `github.token`, I always used `secrets.GITHUB_TOKEN` but it seems to be on both places.

**Comment by GabyCT** [comment]
> Thanks, lgtm now. I was wondering a bit about the `${USER}` but I found out github ignores it whatsoever and simply uses `token` (in documentation they actually mention one can use `Bearer` or `token` but I tried random characters and it's perfectly happy with it. Another good surprise for me was `github.token`, I always used `secrets.GITHUB_TOKEN` but it seems to be on both places.

@ldoktor thanks for all the feedback, I appreciate it

**Review by wainersm — APPROVED** [review]
Thanks @GabyCT !

**Pull request merged**